bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -358,11 +357,71 @@ def _add(self, deployment_pb, bento_pb, bento_path):
                 )
             raise error
 
-    def update(self, deployment_pb):
-        raise NotImplementedError(
-            "Updating AWS Lambda deployment is not supported in current version of "
-            "BentoML"
+    def update(self, deployment_pb, previous_deployment):
+        try:
+            ensure_sam_available_or_raise()
+            ensure_docker_available_or_raise()
+
+            deployment_spec = deployment_pb.spec
+            bento_pb = self.yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    'BentoML currently not support {} repository'.format(
+                        BentoUri.StorageType.Name(bento_pb.bento.uri.type)
+                    )
+                )
+
+            return self._update(
+                deployment_pb, previous_deployment, bento_pb, bento_pb.bento.uri.uri
+            )
+        except BentoMLException as error:
+            deployment_pb.state.state = DeploymentState.ERROR
+            deployment_pb.state.error_message = f'Error: {str(error)}'
+            return ApplyDeploymentResponse(
+                status=error.status_code, deployment_pb=deployment_pb
+            )
+
+    def _update(self, deployment_pb, current_deployment, bento_pb, bento_path):
+        if loader._is_remote_path(bento_path):
+            with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                return self._update(
+                    deployment_pb, current_deployment, bento_pb, local_path
+                )
+        updated_deployment_spec = deployment_pb.spec
+        updated_lambda_deployment_config = (
+            updated_deployment_spec.aws_lambda_operator_config
         )
+        updated_bento_service_metadata = bento_pb.bento.bento_service_metadata
+        try:
+            describe_latest_deployment_state = self.describe(deployment_pb)
+            latest_deployment_state = json.loads(
+                describe_latest_deployment_state.state.info_json
+            )
+            lambda_s3_bucket = latest_deployment_state['s3_bucket']
+
+            _deploy_lambda_function(
+                deployment_pb=deployment_pb,
+                bento_service_metadata=updated_bento_service_metadata,
+                deployment_spec=updated_deployment_spec,
+                lambda_s3_bucket=lambda_s3_bucket,
+                lambda_deployment_config=updated_lambda_deployment_config,
+                bento_path=bento_path,
+            )
+
+            return ApplyDeploymentResponse(
+                deployment=deployment_pb, status=Status.OK()
+            )
+        except BentoMLException as error:
```

Comment:
remove this? I think we already have it at the top level DeploymentOperator interface